src/server/types.ts

```diff
@@ -366,7 +392,7 @@ export abstract class BaseDocumentDriveServer {
     protected abstract deleteDocument(drive: string, id: string): Promise<void>;
 
     /** Event methods **/
-    protected abstract emit<K extends keyof DriveEvents>(
+    abstract emit<K extends keyof DriveEvents>(
```

Comment:
With the mixins approach, the defaultDrivesManager would have access to protected methods like emit, so making it public is not needed.
Otherwise, I'd prefer that the emit method to be passed to the DefaultDrivesManager constructor, to avoid making `emit` public.
This is how the QueueManager and Storage are implemented, they receive a delegate with the methods they need to call from DocumentDriveServer.

I think that the mixins approach will work better, to prevent blocking this PR, let's use this delegate pattern. When ReadMode is merged `DefaultDrivesManager` is a good candidate for the next one to be refactored into a mixin.